group-membership/src/Code.js

```diff
@@ -0,0 +1,151 @@
+};
+
+/**
+ * Installs a trigger in the Spreadsheet to run upon the Sheet being opened.
+ * To learn more about triggers read:
+ * https://developers.google.com/apps-script/guides/triggers
+ */
+function onOpen() {
+  SpreadsheetApp.getUi()
+      .createMenu('Install trigger')
+      .addItem('onEdit', 'installOnEditTrigger')
+      .addToUi();
+}
+
+/**
+ * Installs a trigger in the Spreadsheet that is scheduled
+ * to run upon when values in the Sheet are edited.
+ * To learn more about triggers read:
+ * https://developers.google.com/apps-script/guides/triggers/installable
+ */
+function installOnEditTrigger() {
+  ScriptApp.newTrigger('onEditInstallableTrigger')
+      .forSpreadsheet(SpreadsheetApp.getActive())
+      .onEdit()
+      .create();
+}
+
+/**
+ * Trigger that runs on edit after being installed via the interface.
+ *
+ * @param {Object} e - onEdit trigger event.
+ */
+function onEditInstallableTrigger(e) {
+  // Get the headers, row range and values from the active sheet.
+  const sheet = SpreadsheetApp.getActiveSheet();
+  const headers = sheet.getDataRange().offset(0, 0, 1).getValues()[0];
+  const range = sheet.getRange(e.range.getRow(), 1, 1, headers.length);
+  const row = range.getValues()[0];
+
+  // Convert the row Array into an entries Object using the headers for the
+  // field names.
+  const entries = headers.reduce((result, columnName, i) => {
+    result[columnName] = row[i];
+    return result;
+  }, {});
+
+  // Update the entries Object with the status returned by addToGroup().
+  try {
+    const statusValue = addToGroup(
+        entries[EMAIL],
+        entries[GOOGLE_GROUP],
+        entries[ALLOWED],
+        entries[EMAIL_TEMPLATE_DOC_URL],
+        entries[EMAIL_SUBJECT]
+    );
+    entries[STATUS] = statusValue == STATUS_VALUE.sent ?
+        `${statusValue}: ${new Date()}` : statusValue;
+  } catch (e) {
+    // If there's an error, report that as the status.
+    entries[STATUS] = e;
+  }
+
+  // Convert the updated entries Object into a row Array.
+  const rowToWrite = headers.map((columnName) => entries[columnName]);
+
+  // setValues() receives a 2D array, so we create an array with the row
+  // contents.
+  range.setValues([rowToWrite]);
+}
+
+
+/**
+ * Trigger that runs on edit after being installed via the interface.
+ *
+ * @param {string} userEmail - email of user to add to the group.
+ * @param {string} groupEmail - address of Google Group to add user to.
+ * @param {string} allowed - 'yes' flag to add user to group.
+ * @param {string} emailTemplateDocUrl - Google Doc URL that serves as template
+ *        of the welcome email sent to a user added to the group.
+ * @param {string} emailSubject - subject of welcome email sent to user added
+ *        to group.
+ * @return {string} - status if email was sent to a user added in the sheet.
+ */
+function addToGroup(userEmail, groupEmail, allowed, emailTemplateDocUrl, emailSubject) {
+  if (!allowed) {
+    return STATUS_VALUE.emptyRow;
```

Comment:
Status is already a concept and heavily used so we shouldn't redefine it unless we have a really good reason.
https://developers.google.com/s/results/apps-script?q=getStatus
You can pass or fail this message then log and alert the user of the specific reason
https://developers.google.com/apps-script/guides/dialogs#alert_dialogs

Reply:
Will change the name. Don't really need more logic to the sample.